File: app/src/main/java/su/linka/pictures/components/OutputGrid.java
package su.linka.pictures.components;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.LinearLayout;

import androidx.appcompat.view.menu.MenuBuilder;

import java.util.ArrayList;

import su.linka.pictures.Card;
import su.linka.pictures.Set;

public class OutputGrid extends CardGrid{

    private final ArrayList<Card> cards = new ArrayList<>();

    public OutputGrid(Context context, AttributeSet set) {
        super(context, set);
        setOrientation(LinearLayout.HORIZONTAL);
    }

    public void addCard(Card card){
        cards.add(card);
        render();
    }
    public void backspace(){
        cards.remove(cards.size()-1);
        render();
    }
    public void clear(){
        cards.clear();
        render();
    }

    @Override
    public void setSet(Set set) {
        this.set = set;
        manifest = set.getManifest();
    }
    @Override
    protected void render(){
        removeAllViews();


        int count = cards.size();
        for (int i = 0; i < count; i++) {
            Card card = cards.get(i);
            if(card.cardType!=0) continue;
            GridButton button = new GridButton(getContext()); // Creating an instance for View Object
            button.setLayoutParams(params);

            button.setCard(card);
            button.setImage(set.getBitmap(card.imagePath));
            addView(button, i);

        }
    }
}

File: app/src/main/java/su/linka/pictures/components/OutputGrid.kt
package su.linka.pictures.components

import android.content.Context
import android.util.AttributeSet
import su.linka.pictures.Card
import su.linka.pictures.Set

class OutputGrid @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null
) : CardGrid(context, attrs) {

    private val cards: MutableList<Card> = mutableListOf()

    init {
        orientation = HORIZONTAL
    }

    fun addCard(card: Card) {
        cards.add(card)
        render()
    }

    fun backspace() {
        if (cards.isNotEmpty()) {
            cards.removeAt(cards.lastIndex)
            render()
        }
    }

    fun clear() {
        cards.clear()
        render()
    }

    override fun setSet(set: Set, output: Boolean) {
        this.activeSet = set
        this.manifest = set.getManifest()
        render()
    }

    override fun render() {
        removeAllViews()
        val currentSet = activeSet ?: return
        cards.forEachIndexed { index, card ->
            if (card.cardType != 0) return@forEachIndexed
            val button = GridButton(context, card, currentSet.getBitmap(card.imagePath), true)
            addView(button, index)
        }
    }
}
